tests/unit/python/foglamp/plugins/south/http_south/test_http_south.py

```diff
@@ -0,0 +1,294 @@
+    },
+    'port': {
+        'description': 'Port to listen on',
+        'type': 'integer',
+        'default': '1234',
+    },
+    'host': {
+        'description': 'Address to accept data on',
+        'type': 'string',
+        'default': 'localhost',
+    },
+    'uri': {
+        'description': 'URI to accept data on',
+        'type': 'string',
+        'default': 'sensor-reading',
+    }
+}
+
+
+def mock_request(data, loop):
+    payload = StreamReader(loop=loop)
+    payload.feed_data(data.encode())
+    payload.feed_eof()
+
+    protocol = mock.Mock()
+    app = mock.Mock()
+    headers = CIMultiDict([('CONTENT-TYPE', 'application/json')])
+    req = make_mocked_request('POST', '/sensor-reading', headers=headers,
+                              protocol=protocol, payload=payload, app=app)
+    return req
+
+
+def test_plugin_info():
+    assert http_south.plugin_info() == {
+        'name': 'http_south',
+        'version': '1.0',
+        'mode': 'async',
+        'type': 'south',
+        'interface': '1.0',
+        'config': _DEFAULT_CONFIG(6683)
+    }
+
+
+def test_plugin_init(unused_port):
+    config = _DEFAULT_CONFIG(unused_port())
+    assert http_south.plugin_init(config) == config
+
+
+def test_plugin_start(unused_port):
+    # GIVEN
+    config = copy.deepcopy(_DEFAULT_CONFIG(unused_port()))
+    config['port']['value'] = config['port']['default']
+    config['host']['value'] = config['host']['default']
+    config['uri']['value'] = config['uri']['default']
+
+    # WHEN
+    loop = asyncio.get_event_loop()
+    loop.call_soon(http_south.plugin_start(config))
+
+    # THEN
+    assert config['app'] is not None
+    assert config['handler'] is not None
+    assert config['server'] is not None
+
+
+def test_plugin_reconfigure(mocker, unused_port):
+    # GIVEN
+    config = copy.deepcopy(_DEFAULT_CONFIG(unused_port()))
+    config['port']['value'] = config['port']['default']
+    config['host']['value'] = config['host']['default']
+    config['uri']['value'] = config['uri']['default']
+    mocker.patch.object(http_south, '_plugin_stop', return_value=True)
+    log_exception = mocker.patch.object(http_south._LOGGER, "exception")
+    log_info = mocker.patch.object(http_south._LOGGER, "info")
+
+    # WHEN
+    new_config = http_south.plugin_reconfigure(config, _NEW_CONFIG)
+
+    # THEN
+    assert _NEW_CONFIG == new_config
+    assert 2 == log_info.call_count
+    # TODO: log_info.assert_called_with('Restarting HTTP_SOUTH plugin due to change in configuration keys [uri, port, host]')
+
+
+def test_plugin__stop(mocker, unused_port):
+    # GIVEN
+    config = copy.deepcopy(_DEFAULT_CONFIG(unused_port()))
+    config['port']['value'] = config['port']['default']
+    config['host']['value'] = config['host']['default']
+    config['uri']['value'] = config['uri']['default']
+    log_exception = mocker.patch.object(http_south._LOGGER, "exception")
+    log_info = mocker.patch.object(http_south._LOGGER, "info")
```

Comment:
Assertions like `server.wait_closed()`, `app.shutdown()`, `handler.shutdown(60.0)`, `app.cleanup()` called should be there

Reply:
Not needed in light new logged message. Pl check.